ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -521,17 +538,101 @@ def _run(self) -> Tuple[List[EventOddsRatio], bool]:
         events = positively_correlated_events[:10] + negatively_correlated_events[:10]
         return events, skewed_totals
 
+    def construct_people_url(self, success: bool, event_definition: EventDefinition) -> Optional[str]:
+        #  NOTE: initially only supporting events type, and there is an implicit
+        #  events type when correlation_type is not set.
+        if not self._filter.correlation_type or self._filter.correlation_type == FunnelCorrelationType.EVENTS:
+            # NOTE: we need to convert certain params to strings. I don't think this
+            # class should need to know these details. As we are trying to encode
+            # the filter dict into form encoding, but the persons endpoint is
+            # expecting some attributes to be further encoded as JSON, e.g.
+            # `funnel_correlation_person_entity`, `entities`.
+            # NOTE: I tried using `self._filter.with_data` but this caused errors to
+            # be raised either in `to_dict` or when deserializing on the people endpoint
+            # TODO: remove url serialization details from this class, it likely
+            # belongs in the application layer, or perhaps `FunnelCorrelationPeople`
+            params = self._filter.to_dict()
+            params = {
+                **params,
+                "events": json.dumps(params["events"]),
```

Comment:
this seems incomplete. Since we're dealing with a funnel, we'd want to json dump everything that ought to be dumped in the filters. Off the top of my mind, this means: Exclusion entities, and actions both.

Reply:
I think there is perhaps a method missing on the `filter` object - `def to_params()` which calls `to_dict()` and json-escapes the nested objects?

Reply:
have added a `to_params`